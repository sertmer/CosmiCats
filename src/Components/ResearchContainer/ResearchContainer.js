import React from 'react';
import './ResearchContainer.scss';
import Research from '../Research/Research'

export const ResearchContainer = () => {
  return(
    <section>
      <Research />
    </section>
  )
}

export default ResearchContainer;