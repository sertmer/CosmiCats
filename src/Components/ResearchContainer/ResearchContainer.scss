@import '../../_scss/variables';
@import '../../_scss/mixins';

.research-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 1%;
}

.research-btn {
  height: 10%;
  width: 20%;
  @include rounded-corners(5px);
  margin: 2%;
}

.button-container {
  display: flex;
  flex-direction: column;
  // justify-content: space-between;
  width: 25%;
}

