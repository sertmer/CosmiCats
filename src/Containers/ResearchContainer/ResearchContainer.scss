@import '../../_scss/variables';
@import '../../_scss/mixins';

.research-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 1%;
  height: 90vh;
}

.research-btn {
  height: 10%;
  width: 100%;
  @include rounded-corners(5px);
  margin: 2%;
  background: rgb(69,198,141);
  background: linear-gradient(
    0deg, rgba(69,198,141,1) 0%, 
    rgba(73,210,149,1) 35%, 
    rgba(85,246,175,1) 100%
  );
  font-size: 1.5em;
  font-family: $main-font;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.research-btn:hover {
  transform: scale(1.1);
}


.button-container {
  display: flex;
  flex-direction: column;
  width: 16%;
  justify-content: flex-end;
}



