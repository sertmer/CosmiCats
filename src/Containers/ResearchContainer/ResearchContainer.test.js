import React from 'react';
import { shallow } from 'enzyme';
import { ResearchContainer, mapStateToProps } from './ResearchContainer';

describe('ResearchContainer', () => {

  let wrapper
  let mockNasaImages
  let mockCatImages

  beforeEach(() => {
    mockNasaImages = [
      {
        id: 'ab1',
        img: 'url.com',
        dateCreated: '12-12-12'
      }
    ]

    mockCatImages = [
      {
        id: 'ab2',
        img: 'url2.com',
        dateCreated: '12-13-12'
      }
    ]
    wrapper = shallow(<ResearchContainer 
      nasaImages={mockNasaImages} 
      catImages={mockCatImages}
    />)
  })

  describe('ResearchContainer unit tests', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should increment state.counter when user clicks view next', () => {
      wrapper.setState({counter: 0})
      wrapper.find('.research-btn').simulate('click')

      expect(wrapper.state('counter')).toEqual(1)
    })

    it('should decrement state.counter when user clicks view previous', () => {
      wrapper.setState({counter: 139})
      wrapper.find('.research-btn').simulate('click')

      expect(wrapper.state('counter')).toEqual(138)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with the images', () => {
      const mockState = {
        nasaImages: mockNasaImages,
        catImages: mockCatImages,
        user: 'Davey Havok'
      }

      const expected = {
        nasaImages: mockNasaImages,
        catImages: mockCatImages
      }
      
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})