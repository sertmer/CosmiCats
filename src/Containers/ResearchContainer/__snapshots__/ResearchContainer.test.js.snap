// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResearchContainer ResearchContainer unit tests should match the snapshot 1`] = `
<section
  className="research-container"
>
  <Connect(Research)
    caseStudy={
      Object {
        "dateCreated": "12-13-12",
        "id": "ab2",
        "img": "url2.com",
      }
    }
  />
  <div
    className="button-container"
  >
    <button
      className="research-btn"
      onClick={[Function]}
      type="button"
    >
       view next
    </button>
  </div>
</section>
`;
